File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
void postOrder(struct TreeNode* root, int* ans, int* size) {
    if (!root) return;
    postOrder(root->left, ans, size);
    postOrder(root->right, ans, size);
    ans[(*size)++] = root->val;
}

int* postorderTraversal(struct TreeNode* root, int* returnSize) {
    int* ans = (int*)malloc(100 * sizeof(int));
    *returnSize = 0;
    postOrder(root, ans, returnSize);
    return ans;
}

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    private val ans: MutableList<Int> = mutableListOf();
    private fun postOrd(root: TreeNode?): Unit {
        if (root == null) return;

        postOrd(root.left);
        postOrd(root.right);
        ans.add(root.`val`);
    }
    fun postorderTraversal(root: TreeNode?): List<Int> {
        if (root == null) return emptyList();
        postOrd(root);
        return ans;
    }
}